rs/messaging/src/routing/stream_handler.rs

```diff
@@ -615,14 +615,14 @@ impl StreamHandlerImpl {
                         state,
                         available_guaranteed_response_memory,
                     ) {
-                        Ok(()) => {
+                        None => {
```

Comment:
I just thought about this case here once more. If I'm understanding correctly we might end up dropping responses here that we shouldn't drop once we enable reject signals for requests. 

IIUC `available_guaranteed_response_memory` is not updated when we garbage collect a message from a stream during induction here for simplicity (see the comment on the computation of `available_guaranteed_response_memory`). This means that the memory for a rejected request that we garbage collected before entering this function is not yet added to the available memory yet at this point, which, in turn, seems to suggest that we might end up in situations where we can't push the request here because we're seemingly out of memory.

Am I missing something?

Reply:
I think what I have missed is that `push_input` will ignore available memory for responses given that a response will always free memory.